nf_deployments/v1alpha1/nf_deployment_types.go

```diff
@@ -0,0 +1,201 @@
+/*
+Copyright 2023 Copyright 2023 The Nephio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1alpha1
+
+import (
+	nephioreqv1alpha1 "github.com/nephio-project/api/nf_requirements/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NFDeploymentSpec defines the characteristics of a deployment of a network function
+type NFDeploymentSpec struct {
+	// capacity defines the capacity characteristics of the NF deployment
+	// +optional
+	Capacity *nephioreqv1alpha1.CapacitySpec `json:"capacity,omitempty" yaml:"capacity,omitempty"`
+	// Interfaces defines the interfaces associated with the NF deployment
+	// +optional
+	Interfaces []*InterfaceConfig `json:"interfaces,omitempty" yaml:"interfaces,omitempty"`
```

Comment:
Slices are already pointers, you don't need to make this a slice of pointers. Same is true of maps. The main point is the inbound object needs to be able to specify "nil" vs "empty" - that works already for native pointer types. But for string and int, "empty" means "" and 0, respectively. So, we have to make those pointers to enable us to differentiate between those values and "not sent". Ditto for structs.

Same comment applies throughout these files...I try to note it everywhere I see it but may miss some.

Reply:
fixed